### ISSUE TITLE
Is there a problem with the results?

### ISSUE THREAD

**Issue by QiangHeisenberg** [mozilla/grcov#725]
Q1：There are large sections of code that are not detected, either red or green.

![image](https://user-images.githubusercontent.com/46313511/143675587-128aacb9-add6-43a1-aa25-e08b6468bc87.png)

Q2:Shows that there are four branches here, but there is no response test. But I don't see branches in the code. Am I wrong?

![image](https://user-images.githubusercontent.com/46313511/143675750-668525a9-1fb2-4145-a9cf-6f09c9be7e6d.png)

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cb412e3f2dc101799853216a75604f38

**Comment by corebreaker** [comment]
Yes, me too, block close are covered, it should not be. Maybe cause the droping.

**Comment by marco-c** [comment]
Are you using gcov-based coverage or source-based coverage?

For Q1, it's probably because of `inline`.

**Comment by corebreaker** [comment]
Even if it's caused by something, in any language, block close should not be in coverage result. Closing a block is an element of the structuration of the language.

We cover Rust code not the assembly generated.

**Comment by marco-c** [comment]
Oh actually it is unrelated to inlining, I think it's simply that those functions are only built on big endian and you are building for a little endian target.
Those functions, effectively, are just like comments for your target.

**Comment by marco-c** [comment]
>Even if it's caused by something, in any language, block close should not be in coverage result. Closing a block is an element of the structuration of the language.

There are likely some instructions behind the block close. It's not really related to grcov though, this kind of thing is decided by the implementation of code coverage in the Rust compiler itself.

**Comment by corebreaker** [comment]
Yes there is assembly executed, but , i down't want cover assembly but rust.

**Comment by corebreaker** [comment]
So what a pity that assembly code is covered, and finally i can't use grcov to cover only the Rust code without the assemby.

**Comment by QiangHeisenberg** [comment]
I submitted more detailed information here.https://github.com/rust-lang/rust/issues/91661

**Comment by QiangHeisenberg** [comment]
> Are you using gcov-based coverage or source-based coverage?
> 
> For Q1, it's probably because of `inline`.

I know that the source-based approach does not support branch coverage. However, if branch coverage is not supported, 4 branches shouldn’t be shown here, right?
